lib/printer/loader.js

```diff
@@ -0,0 +1,64 @@
+const path = require('path')
+const fs = require('fs')
+
+const deviceConfig = require('../../device_config.json')
+
+const zebra = require('./zebra')
+const nippon = require('./nippon')
+
+function configureAsZebra (printerCfg) {
+  const address = printerCfg.address
+
+  deviceConfig.kioskPrinter = {
+    maker: 'Zebra',
+    address
+  }
+
+  const jsonDeviceConfig = JSON.stringify(deviceConfig, null, '  ')
+  fs.writeFileSync(path.join(__dirname, '../../device_config.json'), jsonDeviceConfig)
+}
+
+function autoDetect (printerCfg) {
+  // this can be either nippon, zebra or no printer
+  return zebra.checkStatus(printerCfg, 2000)
+    .then(() => {
+      // only zebras can respond
+      configureAsZebra(printerCfg)
+      return zebra
+    })
+    .catch(err => {
+      // can still be nippon, zebra or noPrinter
+      // configure as nippon for the duration of l-m process only
+      // if not nippon printing will fail silently
+      console.log(err)
+      return nippon
+    })
+}
+
+function load () {
+  const printerCfg = deviceConfig.kioskPrinter || { address: '/dev/ttyJ4' }
+  const { maker, model } = printerCfg
+  const noPrinterError = new Error('No printer found')
+
+  if (deviceConfig.brain.mockPrinter) {
+    return Promise.resolve(require('../mocks/kioskPrinter'))
+  }
+
+  // No printer
+  if (maker === 'None') return Promise.reject(noPrinterError)
+
+  // with model  ===  old config version
+  // no maker    ===  gaia shipped without config
```

Comment:
Add more detailed info here on exactly what the issues were with the old shipments.

Something like this but with correct info:

If ``model`` exists, we still need to auto-detect because some shipments were sent out with Nippons but configured for Zebras or vice versa.

If ``maker`` doesn't exist, we still need to auto-detect because some shipments were sent out without a config.